### DISCUSSION TITLE
Come inserire codice Jinja dentro file Markdown?

### DISCUSSION THREAD

**Discussion by cirospat** [opendatasicilia/ods-mkdocs-material#36] [Domande] [answered]
Su mkdocs se in una pagina `.MD` scrivo il seguente codice contenuto tra i 3 backticks di apertura e chiusura

```
{% extends "base.html" %}

{% block htmltitle %}
  <title>Lorem ipsum dolor sit amet</title>
{% endblock %}
```

Poi su pagine `HTML` non mi renderizza il contenuto del testo del codice  perché mi dà un errore.
Non sono riuscito a capire la causa dell'errore.

Pagina `.MD` =
https://github.com/opendatasicilia/ods-mkdocs-material/blob/main/docs/prova-codice.md

Pagina `HTML` =
https://opendatasicilia.github.io/ods-mkdocs-material/prova-codice/

**Comment by aborruso** [comment, marked as answer]
Ciao @cirospat ho messo la risposta in una ricetta: https://opendatasicilia.github.io/ods-mkdocs-material/Ricette/#come-inserire-in-blocchi-di-codice-sintassi-jinja

**Comment by cirospat** [comment]
ho capito (perchè lo hai scoperto tu @aborruso)  che il problema era
un `macro` inserito come plugin nella pagina `mkdocs.yml`
e precisamente alla riga 
https://github.com/opendatasicilia/ods-mkdocs-material/blob/main/mkdocs.yml#L70

Una volta disattivato possiamo editare codice come questo

```
{% extends "base.html" %}

{% block htmltitle %}
  <title>Lorem ipsum dolor sit amet</title>
{% endblock %}
```
dentro una pagina `.MD` e viene renderizzata senza problemi nella pagina `HTML`.

Un giorno capiremo a che serve quel plugin "`macro`".

Grazie tantissimo

**Comment by cirospat** [comment]
ecco
[https://github.com/opendatasicilia/tansignari/issues/183](https://github.com/opendatasicilia/tansignari/issues/183)
come dice @gbvitrano e @aborruso, a cosa serve abilitare il plugin **`macro`** nel file `mkdocs.yml`

teniamone traccia anche qui per non impazzire dopo nel cercare una spiegazione del suo scopo.
